### PR TITLE
Additional unit test related stuff

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

File: lib/common/messages.c [hunk 1]
```diff
@@ -115,11 +115,6 @@ create_reply_adv(const xmlNode *original_request, xmlNode *xml_response_data,
         crm_err("Cannot create new_message, no message type in original message");
         CRM_ASSERT(type != NULL);
         return NULL;
-#if 0
-    } else if (strcasecmp(XML_ATTR_REQUEST, type) != 0) {
-        crm_err("Cannot create new_message, original message was not a request");
-        return NULL;
-#endif
```
Comment (kgaillot) [inline lib/common/messages.c:122]: The idea is that we have request messages and reply messages, and replies should only be generated for requests. Unfortunately in practice we have some odd usages here and there, so we probably can't enforce it. (I know we have issues the other way -- some replies are created as requests.) Maybe change this to a trace message with a comment and don't return.

File: lib/common/nodes.c [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2022 the Pacemaker project contributors
+ * Copyright 2022-2023 the Pacemaker project contributors
```
Comment (kgaillot) [inline lib/common/nodes.c:2]: 2024 throughout :)

File: lib/common/Makefile.am [hunk 3]
```diff
@@ -127,6 +127,7 @@ libcrmcommon_test_la_CFLAGS	= $(libcrmcommon_la_CFLAGS) 	\
 # If -fno-builtin is used, -lm also needs to be added.  See the comment at
 # BUILD_PROFILING above.
 libcrmcommon_test_la_LIBADD	= $(libcrmcommon_la_LIBADD) 	\
+				  -lgcov 			\
```
Comment (kgaillot) [inline lib/common/Makefile.am:130]: We don't require gcc, so you'll have to test availability in configure.ac (or test the --with-coverage option)

**Review by kgaillot — COMMENTED** [review]
nice

File: lib/common/nodes.c (hunk 2 above)
Comment (clumens) [reply lib/common/nodes.c]: Updated.

File: lib/common/Makefile.am (hunk 3 above)
Comment (clumens) [reply lib/common/Makefile.am]: I've added a patch to give us BUILD_COVERAGE to test against, which means that `./configure --with-coverage` requires gcc but we're not enforcing that at configure time.  I could change it to test for the existence of libgcov, however.  I don't think clang ships with that - at least, I don't see it in our packages.

File: configure.ac [hunk 4]
```diff
@@ -533,6 +533,9 @@ AC_ARG_WITH([coverage],
 yes_no_try "$with_coverage" "no"
 with_coverage=$?
 
+AC_DEFINE_UNQUOTED([BUILD_COVERAGE], [$with_coverage], [Build with code coverage])
+AM_CONDITIONAL([BUILD_COVERAGE], [test $with_coverage -ne $DISABLED])
```
Comment (kgaillot) [inline configure.ac:537]: Let's call this `PCMK__WITH_COVERAGE`

Comment (clumens) [reply configure.ac:537]: Both, or just the AM_CONDITIONAL one?

Comment (kgaillot) [reply configure.ac:537]: Only the C one is important, the other can be either way

File: lib/common/messages.c (hunk 1 above)
Comment (clumens) [reply lib/common/messages.c:122]: I can, but it'll still be in a #if 0 block which is why I was looking at removing it in the first place.

Comment (kgaillot) [reply lib/common/messages.c:122]: I mean drop the #if, and have it just log a trace message (not return), with a comment that requiring a match might be a good idea if we can ensure everything complies

**Pull request merged**